### PR TITLE
custom editor configuration option

### PR DESCRIPTION
**Pull request by iloveitaly**
- refactor: slightly better error message
- fix: put header sections which can change first
- feat: support editor config
- docs: helpful docs on overriding existing config and debugging the tool

### CODE REVIEW

**Comment by iloveitaly** [comment]
@laktak friendly reminder here!

**Comment by laktak** [comment]
> fix: put header sections which can change first

Why? I prefer them at the end so the other options do not "move". Please put them back in the old order.

> feat: support editor config

What is your use case? Do you use a different editor than in the terminal or is this something special?

File: README.md [hunk 1]
```diff
@@ -85,6 +85,7 @@ Where `<option>` and `<value>` are one of the specified here:
 | `@extrakto_fzf_tool`        | `fzf`           | Set this to path of fzf if it can't be found in your `PATH`. |
 | `@extrakto_fzf_layout`      |`default`        | Control the fzf layout which is "bottom-up" by default. If you prefer "top-down" layout instead set this to `reverse`. In fact, this value is passed to the fzf `--layout` parameter. Possible values are: `default`, `reverse` and `reverse-list` |
 | `@extrakto_open_tool`       | `auto`          | Set this to path of your own tool or `auto` to use your platforms *open* implementation. |
+| `@extrakto_editor`          | `auto`          | Set this to path of your own editor tool. If not set, `$EDITOR` will be used and `vi` if `$EDITOR1 is not set. |
```
Comment (laur89) [inline README.md:88]: quotation typo

Comment (iloveitaly) [reply README.md:88]: Nice catch!

```suggestion
| `@extrakto_editor`          | `auto`          | Set this to path of your own editor tool. If not set, `$EDITOR` will be used and `vi` if `$EDITOR`is not set. |
```

**Comment by iloveitaly** [comment]
> Why? I prefer them at the end so the other options do not "move". Please put them back in the old order.

On smaller screens this makes it impossible to know what the currently selected option is if you are toggling.

Hmmm what if we made this a config option and flipped the default?

> What is your use case? Do you use a different editor than in the terminal or is this something special?

Yes, I want to use a custom method which parses the input string and cleans it up. In my case, using [this script](https://github.com/iloveitaly/dotfiles/blob/master/.open-file-path.sh) enables me to jump to the file + line + column as opposed to just the file.

**Comment by laktak** [comment]
I've changed `ctrl-` in the help text to `^` and removed the word `area` to make it shorter.  f69af5fb5e5ff6a4ae7e30ec91c7e15146ebdf07 So if your display is at least 75 characters wide you should see the options. Does this help?

Otherwise keep at least insert and copy at the front.

**Comment by iloveitaly** [comment]
I'll give this a shot and let you know!

What do you think about the EDITOR config?

**Comment by laktak** [comment]
> What do you think about the EDITOR config?

Looks good!

File: scripts/extrakto.sh [hunk 2]
```diff
@@ -72,10 +73,12 @@ if [[ "$open_tool" == "auto" ]]; then
...
 
-if [[ -z $EDITOR ]]; then
-    editor="vi" # fallback
-else
-    editor="$EDITOR"
+if [[ -z $editor ]]; then
+    if [[ -z $EDITOR ]]; then
+        editor="vi" # fallback
+    else
+        editor="$EDITOR"
```
Comment (laur89) [inline scripts/extrakto.sh:80]: ```editor="${EDITOR:-vi}"```

File: README.md [hunk 3]
```diff
@@ -158,6 +168,17 @@ optional arguments:
   --warn-empty          warn if result is empty
 ```
 
+# Development
```
Comment (laktak) [inline README.md:171]: I'd prefer if you could add these tips to the wiki instead.

**Pull request merged**